apps/prairielearn/src/pages/workspaceLogs/workspaceLogs.ts

```diff
@@ -115,14 +116,15 @@ async function loadLogsForWorkspaceVersion(
 // Only instructors and admins can access these routes. We don't need to check
 // if the instructor has access to the workspace; that's already been checked
 // by the workspace authorization middleware.
-router.use((req, res, next) => {
-  // TODO: is `authn_is_instructor` the right permission to check?
-  if (!res.locals.authn_is_administrator && !res.locals.authn_is_instructor) {
-    next(new error.HttpStatusError(403, 'Access denied'));
-  } else {
-    next();
-  }
-});
+router.use(
+  asyncHandler(async (req, res, next) => {
+    if (!(await userIsInstructorInAnyCourse({ user_id: res.locals.authn_user.user_id }))) {
```

Comment:
This code maintains the same functionality, though I'm not convinced this is correct. Should this be instructor permission in the course (or the course instance, if there is one; or any course instance if there is none) associated to the current workspace?

Reply:
Changed in 2aed11c.